acq4/drivers/MaiTai/maitai.py

```diff
@@ -0,0 +1,280 @@
+import serial, struct, time, collections, threading, re, pdb
+try:
+    from ..SerialDevice import SerialDevice, TimeoutError, DataError
+except ValueError:
+    ## relative imports not allowed when running from command prompt, so
+    ## we adjust sys.path when running the script for testing
+    if __name__ == '__main__':
+        import sys, os
+        sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+        from SerialDevice import SerialDevice, TimeoutError, DataError
+        
+ErrorVals = {
+    0: ('SP Over-run', 'The previous character was not unloaded before the latest was received.'),
+    1: ('Frame Error', 'A valid stop bit was not received during the appropriate time period.'), 
+    2: ('Buffer Over-run', 'The input buffer is filled and CR has not been received.'),
+    4: ('Bad Command', 'Input can not be interpreted -- command byte not valid.'),
+    8: ('Move Interrupted', 'A requested move was interrupted by input on the serial port.'),
+    16:('Arduino error', 'Error was reported by arduino interface.'),
+    32:('MP285 Timeout', 'Arduino timed out waiting for response from MP285.'),
+    64:('Command timeout', 'Arduino timed out waiting for full command from computer.'),
+}
+    
+
+class TimeoutError(Exception):
+    pass
+
+class MaiTaiError(Exception):
+    pass
+
+class MaiTai(SerialDevice):
+    """
+    Class for communicating with Sutter MP-285 via serial port.
+    
+    Note that this class is NOT thread-safe.
+    """
+    def __init__(self, port, baud=9600):
+        """
+        port: serial COM port (0 => com1)"""
+
+        self.re_float = re.compile(r'\d*\.?\d+')
+        self.port = port
+        self.baud = baud
+        self.sp = serial.Serial(int(self.port), baudrate=self.baud, bytesize=serial.EIGHTBITS,parity=serial.PARITY_NONE,stopbits=serial.STOPBITS_ONE,xonxoff=True)
```

Comment:
Your driver class inherits from SerialDevice but then creates its own second serial port handle.. See the Scientifica driver for an example of a serial device done "correctly".